src/transformers/deepspeed.py

```diff
@@ -36,7 +34,13 @@ def is_deepspeed_available():
     return importlib.util.find_spec("deepspeed") is not None
 
 
-class HfDeepSpeedConfig:
+if is_deepspeed_available():
```

Comment:
The test should probably include `Accelerate` also
```suggestion
if is_accelerate_available() and is_deepspeed_available():
```